web/pdf_viewer.js

```diff
@@ -2256,13 +2297,54 @@ class PDFViewer {
     if (!this.pdfDocument) {
       return;
     }
-    this.#annotationEditorMode = mode;
-    this.eventBus.dispatch("annotationeditormodechanged", {
-      source: this,
-      mode,
-    });
 
-    this.#annotationEditorUIManager.updateMode(mode, editId, isFromKeyboard);
+    const { eventBus } = this;
+    const updater = () => {
+      if (this.#onPageRenderedCallback) {
+        eventBus._off("pagerendered", this.#onPageRenderedCallback);
+        this.#onPageRenderedCallback = null;
+      }
+      if (this.#switchAnnotationEditorModeTimeoutId) {
+        clearTimeout(this.#switchAnnotationEditorModeTimeoutId);
+        this.#switchAnnotationEditorModeTimeoutId = 0;
+      }
```

Comment:
Similarly, this block seems like something that we should also run in the `_resetView` method?